webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/NodeApi.scala

```diff
@@ -562,12 +567,12 @@ class NodeApiService6 (
                        Full(Map[NodeId, Seq[Software]]())
                      }
     } yield {
-      val nodeIds = nodeFilter.getOrElse(nodeInfos.keySet)
       for {
         nodeId    <- nodeIds
         nodeInfo  <- nodeInfos.get(nodeId)
       } yield {
-        serializeInventory(nodeInfo, state, inventories.get(nodeId), software.getOrElse(nodeId, Seq()), detailLevel, version)
+        val runDate = runs.get(nodeId).map( _.map(_.agentRunId.date)).flatten
```

Comment:
flatmap instead of first map + flatten ?

Reply:
done